src/ChannelServer/World/SkillHandler.cs

```diff
@@ -0,0 +1,144 @@
+			Send.ZC_OVERHEAT_CHANGED(caster, skill);
+			if (skill.Data.OverHeat == 0 || skill.Data.OverHeat == skill.OverHeatCount)
+			{
+				Send.ZC_COOLDOWN_CHANGED(caster, skill);
+				skill.OverHeatCount = 0;
+			}
+			var damage = caster.GetRandomPAtk() + 100;
+			Send.ZC_SKILL_MELEE_GROUND(caster, skill, position.X, position.Y, position.Z, null, 0);
+			foreach (var target in targets)
+			{
+				//Send.ZC_SKILL_MELEE_GROUND(caster, skill, target.Position.X, target.Position.Y, target.Position.Z, null, 0);
+				var monster = (Monster)target;
+				if (monster.TakeDamage(damage, caster, 2))
+					Send.ZC_SKILL_CAST_CANCEL(caster, target);
+			}
+		}
+	}
+
+	class GroundSkillHandler : IGroundSkillHandler
+	{
+		public void Handle(Skill skill, Character caster, Position position)
+		{
+			if (skill.Data.OverHeat != 0)
+				skill.OverHeatCount++;
+			Send.ZC_OVERHEAT_CHANGED(caster, skill);
+			if (skill.Data.OverHeat == 0 || skill.Data.OverHeat == skill.OverHeatCount)
+			{
+				Send.ZC_COOLDOWN_CHANGED(caster, skill);
+				skill.OverHeatCount = 0;
+			}
+			Send.ZC_SKILL_READY(caster, skill, caster.Position, position);
+			switch (skill.Id)
+			{
+				case 20007:
+					caster.Map.GetVisibleMonsters(caster);
+					var targets = caster.Map.GetAttackableMonstersInRange(position.X, position.Y, position.Z, (int)skill.Data.MaxRange).ToList<IEntity>();
+					var damage = caster.GetRandomPAtk() + 100;
+					Send.ZC_SKILL_MELEE_GROUND(caster, skill, position.X, position.Y, position.Z, targets, damage);
+					foreach (var target in targets)
+					{
+						Send.ZC_NORMAL_SkillParticleEffect(caster, 1234);
+						Send.ZC_SYNC_START(caster, 1234, 1);
+						Send.ZC_NORMAL_Skill_16(caster, target, position);
+						Send.ZC_SYNC_END(caster, 1234, 0);
+						//Send.ZC_SKILL_MELEE_GROUND(caster, skill, target.Position.X, target.Position.Y, target.Position.Z, null, 0);
+						var monster = (Monster)target;
+						if (monster.TakeDamage(damage, caster, 0))
+							Send.ZC_SKILL_CAST_CANCEL(caster, target);
+					}
+					break;
+				case 30001:
+					Send.ZC_NORMAL_Skill_4E(caster, skill.Id, 1);
+					Send.ZC_NORMAL_Skill(caster, skill, position, caster.Direction, true);
+					Send.ZC_NORMAL_Unknown_06(caster, position);
+					Send.ZC_SYNC_START(caster, 1234, 1);
+					Send.ZC_SYNC_END(caster, 1234, 0);
+					Send.ZC_SYNC_EXEC_BY_SKILL_TIME(caster, 1234, skill.Data.HitDelay);
+					Send.ZC_SKILL_MELEE_GROUND(caster, skill, position.X, position.Y, position.Z, null, 0);
+					Send.ZC_SYNC_EXEC(caster, 1234);
+					// Skill Cast Duration? Cancel's Visible Animation Otherwise you stay in casting animation
+					Send.ZC_SKILL_DISABLE(caster);
+					break;
+				default:
+					//Send.ZC_SYNC_START(caster, 1234, 1);
+					//Send.ZC_SYNC_END(caster, 1234, 0);
+					//Send.ZC_SYNC_EXEC_BY_SKILL_TIME(caster, 1234, skill.Data.HitDelay);
+
+					targets = caster.Map.GetAttackableMonstersInRange(position.X, position.Y, position.Z, (int)skill.Data.MaxRange).ToList<IEntity>();
```

Comment:
If you use curly braces for switch cases you create new scopes and variable names aren't shared between them anymore. This can make it clearer where a variables come from, or rather, that they're created for this case, and aren't shared.